main.go

```diff
@@ -103,8 +104,16 @@ func parseConfigString(conf string) config.Config {
 }
 
 func configuredOptimusTable(s *mgo.Session, table config.Table) optimus.Table {
+	fields := bson.M{}
+	if table.Meta.UseProjectionOptimization == true {
+		// Create a projection to only pull the fields we're interested in
+		for _, f := range table.Fields {
+			fields[f.Source] = 1
+		}
+	}
+
 	collection := s.DB("").C(table.Source)
-	iter := collection.Find(nil).Batch(1000).Prefetch(0.25).Iter()
+	iter := collection.Find(nil).Batch(1000).Prefetch(0.25).Select(fields).Iter()
```

Comment:
`Select` 👍 

I wonder if we should make this `Prefetch` more aggressive. Since we're using streaming IO we could probably make this `> 0.50` since we have multiple go routines that consume data from the iterator (https://godoc.org/labix.org/v2/mgo#Query.Prefetch).